### PR TITLE
chore: Optimization of output path whitelist inspection methods

### PR DESCRIPTION
**Pull request by starhcq**
Optimization of output path whitelist inspection methods

Log: Optimization of output path whitelist inspection methods

Bug: https://pms.uniontech.com/bug-view-236471.html

### CODE REVIEW

File: application/logbackend.cpp [hunk 1]
```diff
@@ -1595,6 +1595,21 @@ bool LogBackend::getOutDirPath(const QString &path)
...
+        QStringList availablePaths =  DLDBusHandler::instance()->whiteListOutPaths();
+        bool bAvailable = false;
+        for (auto path : availablePaths) {
+            if (tmpPath.contains(path)) {
+                bAvailable = true;
+                break;
+            }
+        }
+        if (!bAvailable) {
+            QString titleIcon = ICONPREFIX;
```
Comment (rb-union) [inline application/logbackend.cpp:1608]: 多余代码

File: application/logcollectormain.cpp [hunk 2]
```diff
@@ -323,7 +323,15 @@ void LogCollectorMain::exportAllLogs()
     QFileInfo info(newPath);
     QString outPath = info.path();
     QStringList availablePaths =  DLDBusHandler::instance(this)->whiteListOutPaths();
-    if (!availablePaths.contains(outPath)) {
+
+    bool bAvailable = false;
+    for (auto path : availablePaths) {
+        if (outPath.contains(path)) {
```
Comment (rb-union) [inline application/logcollectormain.cpp:329]: contains的话，路径字符串中可能包含允许路径，建议从起始路径匹配匹配startWith

File: application/logbackend.cpp (hunk 1 above)
Comment (starhcq) [reply application/logbackend.cpp]: 已移除

File: application/logcollectormain.cpp (hunk 2 above)
Comment (starhcq) [reply application/logcollectormain.cpp]: 已修改

**Review by rb-union — COMMENTED** [review]
/+1

**Review by rb-union — APPROVED** [review]
/+1

**Pull request merged**